USER_GUIDE.md

```diff
@@ -169,6 +174,236 @@ for (int i = 0; i < searchResponse.hits().hits().size(); i++) {
+
+CompletionSuggester completionSuggester = FieldSuggesterBuilders.completion()
+                .field("msg")
+                .size(1)
+                .build();
+FieldSuggester fieldSuggester = new FieldSuggester.Builder().prefix("foo")
+                .completion(completionSuggester)
+                .build();
+Suggester suggester = new Suggester.Builder()
+                .suggesters(Collections.singletonMap(suggesterName, fieldSuggester))
+                .build();
+SearchRequest searchRequest = new SearchRequest.Builder()
+                .index(index)
+                .suggest(suggester)
+                .build();
+
+SearchResponse<AppData> response = client.search(searchRequest, AppData.class);
+```
+
+### Using term suggester
+
+```java
+ String index = "term-suggester";
+        
+// term suggester does not require a special mapping
+client.indices().create(c -> c
+                .index(index));
+
+AppData appData = new AppData();
+appData.setIntValue(1337);
+appData.setMsg("foo");
+
+client.index(b -> b
+                .index(index)
+                .id("1")
+                .document(appData)
+                .refresh(Refresh.True));
+
+appData.setIntValue(1338);
+appData.setMsg("foobar");
+
+client.index(b -> b
+                .index(index)
+                .id("2")
+                .document(appData)
+                .refresh(Refresh.True));
+
+String suggesterName = "msgSuggester";
+
+TermSuggester termSuggester = FieldSuggesterBuilders.term()
+                .field("msg")
+                .size(1)
+                .build();
+FieldSuggester fieldSuggester = new FieldSuggester.Builder().text("fool")
+                .term(termSuggester)
+                .build();
+Suggester suggester = new Suggester.Builder()
+                .suggesters(Collections.singletonMap(suggesterName, fieldSuggester))
+                .build();
+SearchRequest searchRequest = new SearchRequest.Builder()
+                .index(index)
+                .suggest(suggester)
+                .build();
+
+SearchResponse<AppData> response = client.search(searchRequest, AppData.class);
+```
+
+### Using phrase suggester
+
+```java
+String index = "phrase-suggester";
+
+String settingsJson = "{\n" +
+                "\"index\": {\n" +
+                "\"analysis\": {\n" +
+                "\"analyzer\": {\n" +
+                "\"trigram\": {\n" +
+                "\"type\": \"custom\",\n" +
+                "\"tokenizer\": \"standard\",\n" +
+                "\"filter\": [\n" +
+                "\"lowercase\",\n" +
+                "\"shingle\"\n" +
+                "]\n" +
+                "}\n" +
+                "},\n" +
+                "\"filter\": {\n" +
+                "\"shingle\": {\n" +
+                "\"type\": \"shingle\",\n" +
+                "\"min_shingle_size\": 2,\n" +
+                "\"max_shingle_size\": 3\n" +
+                "}\n" +
+                "}\n" +
+                "}\n" +
+                "}\n" +
+                "}";
```

Comment:
This is pretty hard to read. Use a multiline Java string?

Reply:
Did one better in the actual tests to use resource files, but updated the example here with a text block.